src/wazuh_db/wdb_parser.c

```diff
@@ -241,18 +242,48 @@ int wdb_parse(char * input, char * output) {
 
         snprintf(sagent_id, sizeof(sagent_id), "%03d", agent_id);
 
-        if (wdb = wdb_open_agent2(agent_id), !wdb) {
-            merror("Couldn't open DB for agent '%s'", sagent_id);
-            snprintf(output, OS_MAXSTR + 1, "err Couldn't open DB for agent %d", agent_id);
-            return -1;
-        }
-
         mdebug2("Agent %s query: %s", sagent_id, query);
 
         if (next = wstr_chr(query, ' '), next) {
             *next++ = '\0';
         }
 
+        if (strcmp(query, "remove") == 0) {
+            snprintf(output, OS_MAXSTR + 1, "ok");
+            result = 0;
+
+            w_mutex_lock(&pool_mutex);
+            if (wdb_remove_database(sagent_id) < 0) {
```

Comment:
Don't we need to close the DB before attempting to remove it? Wouldn't that mean that another thread could still be using it?

Reply:
After discussing it with the team we have decided to use the `wazuhdb remove {id}` query instead of the one used by the agent actor. This is not my preferred approach because it forces us to mantain 2 ways of removing an agent DB, but is the cleaner way we could find to do this.